libgraph/include/katana/PropertyGraph.h

```diff
@@ -114,7 +109,16 @@ class KATANA_EXPORT PropertyGraph {
 
   friend class PGViewCache;
 
-  friend class PropertyGraphRetractor;
+protected:
+  RDG& rdg() { return rdg_; }
+  const RDG& rdg() const { return rdg_; }
+  // TODO(Rob): make these private. Would be good to have protected accessor
+  //            but there is already a public const accessor, so a protected
+  //            non-const accessor throws compile errors
```

Comment:
Why does a non-const acessor throw compile errors?

`void Fn() const` and `void Fn()` are considered different functions: https://godbolt.org/z/vdqEKM3jn

Reply:
Agree with Donald. Can't we have a non-const accessor for these variables instead?

Reply:
It doesn't have to do with inheritance. Here is what happens: https://godbolt.org/z/8xrnaoTox

Reply:
I see. Having accessors with the same name at different protection levels is somewhat annoying.

You'd have to do something like: https://godbolt.org/z/eM91aoa6M to disambiguate these calls. Or you can simply give the mutating accessor a different name. It is protected anyway, subclasses should be ok with a different name for it.

Reply:
Yeah I don't think forcing people using the public API to const-cast when they want to call that method is a great solution haha. I will just make the protected method `mutable_<var>` instead of `<var>` so there is no ambiguity.